tests/test_inngest/test_client/test_send.py

```diff
@@ -130,6 +132,229 @@ async def test_partial_send_error(self) -> None:
+                    app_id=random_suffix("my-app"),
+                    is_production=False,
+                )
+
+                event_name = random_suffix("foo")
+                events = [
+                    inngest.Event(name=event_name),
+                    inngest.Event(name=event_name),
+                ]
+                if is_sync:
+                    client.send_sync(events)
+                else:
+                    await client.send(events)
+
+                # client didn't retry request
+                assert proxy_request_count == 1
+
+
+CountGetter = typing.Callable[[], int]
+
+
+# Create a proxy that mimics a request reaching the Dev Server but the
+# client receives a 500 on the first attempt. This ensures that the Dev
+# Server's event processing logic properly handles the idempotency key
+# header.
+def create_first_request_500_handler() -> typing.Tuple[
+    http_proxy.OnRequest, CountGetter
+]:
+    proxy_request_count = 0
+
+    def on_request(
+        body: typing.Optional[bytes],
+        headers: dict[str, list[str]],
+        method: str,
+        path: str,
+    ) -> http_proxy.Response:
+        nonlocal proxy_request_count
+        proxy_request_count += 1
+
+        # Always forward request to real dev server
+        resp = httpx.request(
+            method=method,
+            url=f"{dev_server.server.origin}{path}",
+            headers={k: v[0] for k, v in headers.items()},
+            content=body,
+        )
+
+        # But make client think we failed with synthetic 500 response on first try
+        if proxy_request_count == 1:
+            return http_proxy.Response(
+                body=b"{}",
+                headers={},
+                status_code=500,
+            )
+        else:
+            # forward subsequent dev server responses
+            return http_proxy.Response(
+                body=resp.content,
+                headers=dict(resp.headers),
+                status_code=resp.status_code,
+            )
+
+    def get_count() -> int:
+        return proxy_request_count
+
+    return on_request, get_count
+
+
+# Create a proxy that mimics a request reaching the Dev Server but the
+# client times out on the first attempt.
+def create_first_request_timeout_handler() -> typing.Tuple[
+    http_proxy.OnRequest, CountGetter
+]:
+    proxy_request_count = 0
+
+    def on_request(
+        body: typing.Optional[bytes],
+        headers: dict[str, list[str]],
+        method: str,
+        path: str,
+    ) -> http_proxy.Response:
+        nonlocal proxy_request_count
+        proxy_request_count += 1
+
+        # Always forward request to real dev server
+        resp = httpx.request(
+            method=method,
+            url=f"{dev_server.server.origin}{path}",
+            headers={k: v[0] for k, v in headers.items()},
+            content=body,
+        )
+
+        # But make client think we timed out
+        if proxy_request_count == 1:
+            time.sleep(35)
```

Comment:
this sucks and makes the test unnecessarily long, should I add a variable for https://github.com/inngest/inngest-py/blob/063812d75358a2b90422e205ed00480abb9b95ed/pkg/inngest/inngest/_internal/client_lib/client.py#L181 that we set in this test? Or should we even consider elevating `timeout` to a constructor argument of the `Inngest` client?

Reply:
discussed in slack with @amh4r , we're adding a `request_timeout` argument